frontend/src/pages/home/learn/planning-to-save-for-retirement.tsx

```diff
@@ -0,0 +1,126 @@
+import { FC } from 'react'
+
+import { GetServerSideProps } from 'next'
+import { useTranslation, Trans } from 'next-i18next'
+import { serverSideTranslations } from 'next-i18next/serverSideTranslations'
+import { NextSeo } from 'next-seo'
+import Image from 'next/image'
+import {GoTriangleUp} from 'react-icons/go'
+import {MdArrowForwardIos} from 'react-icons/md'
+
+import { useEffect, useState} from 'react'
+
+const PlanningToSaveForRetirement: FC = () => {
+  const { t } = useTranslation('learn/planning-to-save-for-retirement')
+
+  // TODO -- side page navigation to change background colour of nav item on page scroll
+  const [offset, setOffset] = useState(0);
+  useEffect(() => {
+      const onScroll = () => setOffset(window.pageYOffset);
+      // clean up listeners:
+      window.removeEventListener('scroll', onScroll);
+      window.addEventListener('scroll', onScroll, { passive: true });
+      return () => window.removeEventListener('scroll', onScroll);
+  }, []);
+
+  useEffect(()=>{
+    let mainSections = document.querySelectorAll('#content h2')
+    let navItems = document.querySelectorAll('#page-nav a')
+    let intersection = navItems[[...mainSections].findIndex(section=>section.getBoundingClientRect().top-10>=0)-1]
+    navItems.forEach(item=>item===intersection ? item.classList.add('page-nav-active') : item.classList.remove('page-nav-active'))
+  },[offset])
+  
+
+  return (
+    <div>
+      <NextSeo title={t('header')} />
+      <h1 className="h1 bg-gray-100 p-4 md:p-16 rounded-3xl">{t('header')}</h1>
+      <div className='pt-2 md:flex gap-10'>
+        <section className='hidden md:block md:flex-[0.5]'>
+          <nav id='page-nav' className="border-2 rounded-lg shadow-sm shadow-black text-[.9em] sticky top-0">
+            <h2 className='text-[1.1em] p-5 font-bold border-b-2'>{t('header')}</h2>
+            <a href="#overview" className='block no-underline p-5 text-black visited:text-black hover:bg-[#4ED8E8] hover:bg-opacity-[12%] hover:text-[#008490]'>{t('overview-link-text')}</a>
```

Comment:
This can be dynamically rendered.
```tsx
[
{ id: "how-much-will-you-need", text: t('how-much-will-you-need-heading') },
{ id: "changes-with-age", text: t('changes-with-age-heading') },
].map(({id, text}) => (
<a href={`#${id}`} className='block no-underline p-5 text-black visited:text-black hover:bg-[#4ED8E8] hover:bg-opacity-[12%] hover:text-[#008490]'>{text}</a>
))
```